compiler/noirc_frontend/src/tests.rs

```diff
@@ -2729,3 +2729,23 @@ fn incorrect_generic_count_on_type_alias() {
     assert_eq!(actual, 1);
     assert_eq!(expected, 0);
 }
+
+#[test]
+fn uses_self_for_struct_function_call() {
+    let src = r#"
+    struct S { }
+
+    impl S {
+        fn one() -> Field {
+            1
+        }
+
+        fn two() -> Field {
+            Self::one() + Self::one()
```

Comment:
Please add a test to access a struct member from `Self` to make sure those resolve properly, e.g. an impl method `fn foo(self) { self.x + self.y }` .. `fn main() { assert(something.foo() == expected) }`.

And also perhaps the other cases discussed in slack:
```rust
trait Foo {
  fn foo() {}
}

impl Foo for T { } // <- T here

// OR
trait Foo where Self: EQ { // <- Self here
  fn foo(self) -> bool {
    self == self
  }
}

// OR
trait Foo {
  fn foo() {
    Self::bar() // <-- this Self
  }
}

Reply:
> e.g. an impl method fn foo(self) { self.x + self.y }

I think that goes through a different code path, so maybe the changes in this PR don't affect that. For example there's a test that uses `self.a`:

https://github.com/noir-lang/noir/blob/cca89c10b476fbf500dcf471c7b2271717a6cf61/compiler/noirc_frontend/src/tests.rs#L129

> And also perhaps the other cases discussed in slack

👍

Reply:
Done!